rts/Lua/LuaSyncedRead.cpp

```diff
@@ -4150,6 +4151,27 @@ int LuaSyncedRead::GetUnitCosts(lua_State* L)
 	lua_pushnumber(L, unit->cost.energy);
 	return 3;
 }
+/***
+ * @function Spring.GetUnitCostTable
+ * @number unitID
+ * @treturn nil|{ metal = number, energy = number }, number buildTime
+ * @treturn number buildTime
+ */
+int LuaSyncedRead::GetUnitCostTable(lua_State* L)
+{
+	const CUnit* const unit = ParseInLosUnit(L, __func__, 1);
+	if (unit == nullptr)
+		return 0;
+	lua_createtable(L, 2, 1);
+	lua_pushnumber(L, unit->cost.metal);
+	lua_pushstring(L, "metal");
```

Comment:
The value should be on top of the stack (i.e. added second), and the key should be the 2nd from the top (i.e. added first)
```suggestion
	lua_pushstring(L, "metal");
	lua_pushnumber(L, unit->cost.metal);
```

Reply:
Done